### ISSUE TITLE
Behavior annex parser flags the entire annex block on syntax errors

### ISSUE THREAD

**Issue by smithdtyler** [osate/osate2#1863]
If I make a small error like swapping a colon and semicolon, OSATE flags the _entire_ BA block as being an error. 

<img width="491" alt="Screen Shot 2019-06-27 at 2 47 25 PM" src="https://user-images.githubusercontent.com/8142684/60296004-b7818a00-98ea-11e9-94fe-9b891ba90e68.png">

```aadl
  thread group implementation threadB.impl
    subcomponents
      t0 : thread;
      annex behavior_specification {**
    states
    s0: initial complete final state;
    s1; state;
    transitions
    s0 -[ on dispatch ]-> s1 {output:=1};
    **};
  end threadB.impl;
```

The error message is unhelpfully complicated:

> The required feature 'initialState' of 'org.osate.ba.aadlba.impl.BehaviorAnnexImpl@46462898{platform:/resource/com.adventiumlabs.sliced.unittest/LivenessAssertionFail.aadl#/0/@ownedPublicSection/@ownedClassifier.4/@ownedAnnexSubclause.0/@parsedAnnexSubclause}' must be set

## Environment

* **OSATE Version**: 2.5.0 
* **Operating System**: Windows 7

**Comment by Etienne13** [comment]
This problem is due to the fact the parsed annex object is kept assigned to the result provided by the parser even though the parser reported errors. In class AnnexParserAgent, line 225, you can see that the second part of the conjunction has been commented. If you assign the parsed annex object to null when this condition is true (which is the case in your example) the strange error message disappears.

However, syntax highlighting will then be off (see #1864) and no error is reported (probably same issue as #891).

**Comment by lwrage** [comment]
Closed via PR #1875
